### PR TITLE
Routing api functionalities

### PR DESCRIPTION
**Pull request by ghost**
Hello! This is my React/Redux module capstone project. 

This is the Loom video presenting it briefly and a piece of code I am proud of: https://www.loom.com/share/3ad49050dbe040b3bb879264e400df31

The requirements I followed were those:
First you need to choose an API to base the development of the webapp on. The API should allow you to get numeric values based on a parameter. We recommend that you choose an API that is open or needs an API key. If you choose an API that require authentication, you should implement it on your own.

Some example APIs are:

    [Financial modeling prep API](https://financialmodelingprep.com/developer/docs/): stock data.
    [Narrativa API](https://covid19tracking.narrativa.com/index_en.html): COVID-19 data.
    [Air pollution API](https://openweathermap.org/api/air-pollution): air quality data.

You can find many APIs in [this GitHub repo](https://github.com/public-apis/public-apis) or in [ProgrammableWeb](https://www.programmableweb.com/category/all/apis). Some of the APIs requires a token for authentication and some others are just open.

Design

    You should follow these [design guidelines](https://www.behance.net/gallery/31579789/Ballhead-App-(Free-PSDs)), including:
        Colors (select one main color for your website).
        Typography: font face, size and weight.
        Layout: composition and space between elements.

Original design idea by [Nelson Sakwa on Behance](https://www.behance.net/sakwadesignstudio).

The [Creative Commons license of the design](https://creativecommons.org/licenses/by-nc/4.0/)[](https://github.com/microverseinc/curriculum-react-redux/blob/main/capstone/react_capstone.md#project-documentation) requires that you give appropriate credit to the author. Therefore, you must do it in the README of your project.

Interactions

    Home page
        When the page loads, the webapp shows the list of categories that could be filtered out by some parameter, for example by the category name.
        Along with the category name, you will display some numeric values per category that come from the API.
        When the user clicks (or taps) on a category item, the application navigates to the details page.

    Details page
        In the details page, the webapp retrieves data from the API to show detailed data in the category.
        When the user clicks on the "Back" button (<), the user navigates to the home page.

Testing requirements

    Create unit tests for pure functions in your Redux code.
    Create integration tests for your application using the React Testing Library.
        You may need to mock the access to the API, so that your tests don't send actual requests.
        You may need to mock the connection to the Redux Store.

Technical requirements

    The project is a single-page application (SPA) built with React and Redux.
    The data retrieved from the API should be stored in the Redux store.
    You should filter the data that you retrieve from the API using a Filter stateless component.
    Every page (the main page and the pages for each item) should have a unique route within the SPA.
    The project should be deployed and accessible online (add a link to the online version of your app to your README file).

Project documentation

Once you have finished the development of the project, you should record a video presenting the features of the project you built. It is a video with a maximum length of 5 minutes. The content of the video should include:

    a description of the project.
    a demo of the project features.
    you should also highlight some interesting piece of code or something you built that you are very proud of.

For recording the video you can use tools like [Loom](https://www.loom.com/) that let you share a private link to the recording, and configure a shot that shows your computer screen and your face at the same time in a small picture.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,70 +1,45 @@
...
+## Author
 
-The build is minified and the filenames include the hashes.\
-Your app is ready to be deployed!
+### 👤 Ivan Silva
+- GitHub: [iKuartz](https://github.com/iKuartz)
+- LinkedIn: [Ivan Barbosa da Silva](https://www.linkedin.com/in/ivan-silva-a47058b3/)
 
-See the section about [deployment](https://facebook.github.io/create-react-app/docs/deployment) for more information.
+## Contributing
```
Comment (SajjadAhmad14) [inline README.md:38]: Please add the `Acknowledgements` section or similar in README that gives appropriate credit to
the design author.

File: src/index.js [hunk 2]
```diff
@@ -1,17 +1,18 @@
...
 import './index.css';
 import App from './App';
-import reportWebVitals from './reportWebVitals';
 
 ReactDOM.render(
   <React.StrictMode>
-    <App />
+    <Provider store={store}>
+      <BrowserRouter>
+        <App />
```
Comment (SajjadAhmad14) [inline src/index.js:13]: The app shows a blank page so I am unable to test the functionality and design. Please make sure you are creating a pull request from the branch where the project code resides.
blank

File: README.md [hunk 3]
```diff
@@ -1,70 +1,45 @@
-# Getting Started with Create React App
+# Covid-Tracker
+A Covid Tracker Webapp with a mobile-first design that displays the present number of confirmed cases of Covid across selected countries, incorporating the Narrative API to a React-Redux framework.
 
-This project was bootstrapped with [Create React App](https://github.com/facebook/create-react-app).
+<!-- ![screenshot](src/assets/screenshot.png) -->
 
-## Available Scripts
+## Live Demo
```
Comment (SajjadAhmad14) [inline README.md:6]: Please deploy the project and add a link to the deployed project in the README.

File: src/App.test.js [hunk 4]
```diff
@@ -1,8 +0,0 @@
-import { render, screen } from '@testing-library/react';
-import App from './App';
-
```
Comment (SajjadAhmad14) [inline src/App.test.js:3]: Please add unit tests for components of the app and also for some pure functions in Redux.
Please add integration tests using `React Testing Library`

**Review by SajjadAhmad14 — CHANGES_REQUESTED** [review]
## Required Changes ♻️
> "Change is inevitable. Change is constant. -Benjamin Disraeli"
## Hi Ivan, 
 ### You have been doing a tremendous job so far :ok_hand:

## To Highlight 🎉
PR has a title and description. :heavy_check_mark:
The project presentation is impressive. :heavy_check_mark:
## To Improve 🙏

- [x] Please check inline comments under the review

## Optional suggestions
_Comments with the [OPTIONAL] prefix are not a must to implement however, It's strongly recommended to work on them as they can make your code better._

## Happy coding!👏👏👏

#### Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
#### Please remember to tag me in your question so I can receive the notification.
**_Note: Please do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

File: README.md (hunk 1 above)
Comment (ghost) [reply README.md:38]: Totally right, forgot that completely, already corrected.

File: src/index.js (hunk 2 above)
Comment (ghost) [reply src/index.js:13]: This was a result of my wrong implementation of GitHub Pages. Now it works on Netlify.

File: README.md (hunk 3 above)
Comment (ghost) [reply README.md:6]: Deployed on Netlify, and added a link to the preview to the readme. The final address will be available when this pull request goes to the production branch.

File: src/App.test.js (hunk 4 above)
Comment (ghost) [reply src/App.test.js:3]: I completely forgot the tests. Sorry, will add them right away.

File: src/__mocks__/configureStore.js [hunk 5]
```diff
@@ -0,0 +1,15 @@
+import { createStore, combineReducers, applyMiddleware } from 'redux';
+import logger from 'redux-logger';
+import thunk from 'redux-thunk';
+import countriesReducer from './countries';
+
```
Comment (SajjadAhmad14) [inline src/__mocks__/configureStore.js:5]: All the tests are failing. Kindly see the screenshots below and make sure all the tests are passing.
tests

File: src/__tests__/components/CountriesContainer.test.js [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+import store from '../../__mocks__/configureStore';
+import CountriesContainer from '../../components/CountriesContainer';
+
+describe('Countries snapshot', () => {
+  test('Check if renders correctly', () => {
+    const tree = renderer
+      .create(
+        <Provider store={store}>
+          <BrowserRouter>
+            <CountriesContainer />
```
Comment (SajjadAhmad14) [inline src/__tests__/components/CountriesContainer.test.js:14]: Please use `grill sans` font for the body and title on both pages. Kindly use the same typography as in the design for all the pages.

File: src/__tests__/components/Navigation.test.js [hunk 7]
```diff
@@ -0,0 +1,21 @@
...
+import store from '../../__mocks__/configureStore';
+import Navigation from '../../components/Navigation';
+
+describe('CNavigation snapshot', () => {
+  test('renders correctly', () => {
+    const tree = renderer
+      .create(
+        <Provider store={store}>
+          <BrowserRouter>
+            <Navigation />
```
Comment (SajjadAhmad14) [inline src/__tests__/components/Navigation.test.js:14]: Please add a `Back` button on the `Details` page which allows the user to go to the `Home` page.

**Review by SajjadAhmad14 — CHANGES_REQUESTED** [review]
## Required Changes ♻️
> "Change is inevitable. Change is constant. -Benjamin Disraeli"
## Hi Ivan,
### Great work on making the changes requested by a previous reviewer 👏 
### You've done well implementing some of the requested changes, but there are still some which aren't been addressed yet.
## To Improve 🙏

- [ ] Please check inline comments under the review

## Optional suggestions
_Comments with the [OPTIONAL] prefix are not a must to implement however, It's strongly recommended to work on them as they can make your code better._

## Happy coding!👏👏👏

#### Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
#### Please remember to tag me in your question so I can receive the notification.
**_Note: Please do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

File: src/__mocks__/configureStore.js (hunk 5 above)
Comment (ghost) [reply src/__mocks__/configureStore.js:5]: image
Sir, I changed nothing on the code and all tests pass on my machine. 
I also see no screenshots attached to your comment.

File: src/__tests__/components/CountriesContainer.test.js (hunk 6 above)
Comment (ghost) [reply src/__tests__/components/CountriesContainer.test.js:14]: Ok, changed that.

File: src/__tests__/components/Navigation.test.js (hunk 7 above)
Comment (ghost) [reply src/__tests__/components/Navigation.test.js:14]: Ok, done.

**Review by Cathella — APPROVED** [review]
## 🔥 Status: Approved 🟢

### Hi @iKuartz 👋🏻, 

Great job making the requested changes 💪🏻! Your project looks really nice 😍.
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**